Zookeeper + clustered event bus configuration example

mis94 opened the pull request:
An example application using Zookeeper as cluster manager showing configuration, also shows how to configure event bus in order to work appropriately on multiple machines.

cescoffier commented on zookeeper-examples/README.adoc:
```diff
@@ -0,0 +1,19 @@
+# Vertx-Clustered-EventBus-with-Zookeeper
+
```
Can add a few sentences explaining how to run the example. Typically how do I start the Zookeeper cluster, how do I run the example.

mis94 replied:
there is a section in the end of the readme file explaining how to run on a single machine and on multiple machines, do you mean explaining how to run Zookeeper? If so I can add more details about how do I run Zookeeper on my machine

cescoffier replied:
Yes exactly. Typically I use Docker to run my Zookeeper server on my dev machine. I don't know what's the best: using docker or explaining how to install it and run it.

cescoffier commented on zookeeper-examples/pom.xml:
```diff
@@ -0,0 +1,70 @@
+    </build>
+
+    <properties>
+        <java.version>1.8</java.version>
```
Is this property required?

mis94 replied:
No, seems to be unused. I will remove it.

cescoffier commented on zookeeper-examples/pom.xml:
```diff
@@ -0,0 +1,70 @@
+
+    <properties>
+        <java.version>1.8</java.version>
+        <vertx.version>3.5.0</vertx.version>
```
Should be bumped to 3.5.1

mis94 replied:
Ok, I will upgrade the Vert.x version.

cescoffier commented on zookeeper-examples/src/main/java/MainClassNode1.java:
```diff
@@ -0,0 +1,61 @@
+    private static JsonObject configureClusterManager() {
+        // add the IP of the machine hosting the cluster manager
+        JsonObject zkConfig = new JsonObject();
+        zkConfig.put("zookeeperHosts", "192.168.1.12");
```
Can't it be `localhost`

cescoffier replied:
Ok, but for the user running the example on a laptop, how would that work? Editing the code to enter the IP is not very user-friendly. We can add a comment explaining how to configure it for more "production" use cases.

mis94 replied:
Do you prefer comments in code, or add a section in readme?

cescoffier replied:
Good question. I don't have a strong opinion. I like having things in the code so the user can quickly understand where to _patch_ during a copy-paste (as we all do)

mis94 replied:
Ok, I will add these modifications and explain how to run Zookeeper and update the PR.

cescoffier commented on zookeeper-examples/src/main/java/MainClassNode1.java:
```diff
@@ -0,0 +1,61 @@
+        // else if working on different separate machines it doesn't matter
+        VertxOptions options = new VertxOptions()
+                .setClustered(true)
+                .setClusterHost("192.168.1.12")
```
Why is the clusterHost required?

cescoffier replied:
Ok, so we should remove it and add a comment about it.

cescoffier commented on zookeeper-examples/src/main/java/MainClassNode1.java:
```diff
@@ -0,0 +1,61 @@
+        VertxOptions options = new VertxOptions()
+                .setClustered(true)
+                .setClusterHost("192.168.1.12")
+                .setClusterPort(17001)
```
Why is the cluster port required?

cescoffier replied:
Ok, so we should remove it and add a comment about it.

cescoffier commented on zookeeper-examples/src/main/java/MainClassNode2.java:
```diff
@@ -0,0 +1,61 @@
+    private static JsonObject configureClusterManager() {
+        // add the IP of the machine hosting the cluster manager
+        JsonObject zkConfig = new JsonObject();
+        zkConfig.put("zookeeperHosts", "192.168.1.12");
```
Can't we use localhost?

cescoffier commented on zookeeper-examples/src/main/java/MainClassNode2.java:
```diff
@@ -0,0 +1,61 @@
+        // else if working on different separate machines it doesn't matter
+        VertxOptions options = new VertxOptions()
+                .setClustered(true)
+                .setClusterHost("192.168.1.56")
```
Why is clusterHost and clusterPort required.

mis94 commented:
@cescoffier I've updated in the PR with the following:
- Removed the IPs and ports from code and added clarifying comments instead so that the example can easily run on user's machine without any modifications.
- Added some files necessary for running Zookeeper server on user's machine (this is the way I run Zookeeper server on my machine).
- Added a new section to readme file explaining how to run Zookeeper server and run the project.

cescoffier commented on zookeeper-examples/Zookeeper/config/log4j.properties:
```diff
@@ -0,0 +1,92 @@
+log4j.logger.org.apache.kafka=INFO
+
+# Change to DEBUG or TRACE to enable request logging
+log4j.logger.kafka.request.logger=WARN, requestAppender
```
Why do we have kafka related configuration?

cescoffier commented on zookeeper-examples/Zookeeper/config/log4j.properties:
```diff
@@ -0,0 +1,92 @@
+
+# Access denials are logged at INFO level, change to DEBUG to also log allowed accesses
+log4j.logger.kafka.authorizer.logger=INFO, authorizerAppender
+log4j.additivity.kafka.authorizer.logger=false
```
Same here

mis94 commented:
I just added the way I'm running Zookeeper server on my machine, but the user is free to start the Zookeeper server the way he wants. Do you mean we should erase these files and just mention the ways they can run Zookeeper server and add links showing how to do so in the readme?

mis94 commented:
@cescoffier Any news?

mis94 commented:
@cescoffier I've updated the PR with the following:
- As you suggested, I used Docker for running Zookeeper. I wasn't using Docker at the time I made the PR as I wasn't familiar with it then.
- Removed all unneeded Kafka configuration files that I used before to run Zookeeper.

cescoffier commented:
Thanks!
